source/crc.c

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include "checksums.h"
+
+#include "aws/checksums/crc.h"
+#include "aws/common/byte_buf.h"
+
+PyObject *aws_py_checksums_crc32(PyObject *self, PyObject *args) {
+    (void)self;
+
+    Py_buffer input_buf;
+    PyObject *py_previousCrc32;
+
+    if (!PyArg_ParseTuple(args, "s*O", &input_buf, &py_previousCrc32)) {
+        return PyErr_AwsLastError();
+    }
+
+    uint32_t previousCrc32 = PyLong_AsUnsignedLong(py_previousCrc32);
+
+    if (previousCrc32 == (uint32_t)-1 && PyErr_Occurred()) {
+        return NULL;
+    }
+
+    // struct aws_byte_buf input = aws_byte_buf_from_array(input_buf.buf, input_buf.len);
+
+    // uint32_t crc_res = aws_checksums_crc32(input.buffer, input.len, previousCrc32);
+
+    uint32_t crc_res = aws_checksums_crc32(input_buf.buf, input_buf.len, previousCrc32);
+
```

Comment:
Py_buffer.buf isn't necessarily a contiguous block of memory. [Docs](https://docs.python.org/3/c-api/buffer.html#c.Py_buffer):
> For contiguous arrays, this is the length of the underlying memory block. For non-contiguous arrays, it is the length that the logical structure would have if it were copied to a contiguous representation

Python's built-in CRC function [checks if the buffer is contiguous](https://github.com/python/cpython/blob/ff230153fd2b1d0576df21ae6c29f0f2fdb0244f/Modules/clinic/zlibmodule.c.h#L760), we should too.

Reply:
Should I just set the argument to bad argument and return null if that's the case, like what they do in the example?